src/mss/windows.py

```diff
@@ -253,6 +294,67 @@ def _grab_impl(self, monitor: Monitor, /) -> ScreenShot:
 
         return self.cls_image(bytearray(self._handles.data), monitor)
 
-    def _cursor_impl(self) -> Optional[ScreenShot]:
-        """Retrieve all cursor data. Pixels have to be RGB."""
-        return None
+    def _cursor_impl(self) -> ScreenShot:
+        """Retrieve all cursor data. Pixels have to be RGB.
+
+        [1] user32.DrawIcon(HDC(memdc), 0, 0, hcursor)
+        Sometimes the memdc value is greater than the 32 bit limit
+        and that results in
+        'ctypes.ArgumentError: argument 1: OverflowError: int too long to convert'
+        but casting it to HDC type seems to fix the issue.
+
+        [2] user32.GetIconInfo(hcursor, self._handles.iconinfo)
+        GetIconInfo also returns the handle for mask bitmap and the handle for color bitmap
+        but the color bitmap handle is null in case of monochrome cursors.
+
+        [3] is_monochrome = self._handles.iconinfo.hbmColor is None
+        The correct way to detect monochrome cursors seems to be a unique property of their
+        mask bitmap. The height of the mask bitmap of a monochrome cursor is twice its width
+        (https://learn.microsoft.com/en-us/windows/win32/api/winuser/ns-winuser-iconinfo)
+        But I cannot find the correct way of getting a bitmap's dimensions, therefore I
+        just eneded up checking if the color bitmap is null.
+
+        [4]
+        The data received using DrawIcon is in the format BGRA but in case of monochrome
+        cursors the alpha value of every pixel is 0 for some reason. Therefore, the alpha
+        value of every non black pixel has to be manually set to 255.
+        """
+        srcdc, memdc = self._handles.srcdc, self._handles.memdc
```

Comment:
i did not care to look into it but now that you said it i will see if something can be done about simplifying it to just the cursor
i will update you once im done

Reply:
UPDATE: im afraid it would not be possible since the cursor is a part of the screen's device context, it should not be a huge issue as im making a small 32x32 bitmap and capturing only the cursor's image
im sorry if im wrong, im not an expert at windows api and i would love to learn something new